### PR TITLE
Change way browser and version found

### PR DESCRIPTION
**Pull request by squirrelo**
There was a problem with firefox and opera where the version found was not the actual version of the browser. This new code fixes that.

### CODE REVIEW

File: qiita_pet/templates/sitebase.html [hunk 1]
```diff
@@ -34,37 +34,20 @@
...
-        if ((ix=objfullVersion.indexOf(";"))!=-1)
-          objfullVersion=objfullVersion.substring(0,ix);
-        if ((ix=objfullVersion.indexOf(" "))!=-1)
-          objfullVersion=objfullVersion.substring(0,ix);
-        objBrMajorVersion = parseInt(''+objfullVersion,10);
-        if (isNaN(objBrMajorVersion)) {
-          objfullVersion = ''+parseFloat(navigator.appVersion);
-          objBrMajorVersion = parseInt(navigator.appVersion,10);
-        }
+        // from http://stackoverflow.com/questions/5916900/how-can-you-detect-the-version-of-a-browser
```
Comment (ElDeveloper) [inline qiita_pet/templates/sitebase.html]: Can you use the permanent link instead? http://stackoverflow.com/q/5916900/379593

File: qiita_pet/templates/sitebase.html [hunk 2]
```diff
@@ -34,37 +34,20 @@
...
+        var ua=navigator.userAgent,tem,M=ua.match(/(opera|chrome|safari|firefox|msie|trident(?=\/))\/?\s*(\d+)/i) || []; 
+    if(/trident/i.test(M[1])){
```
Comment (ElDeveloper) [inline qiita_pet/templates/sitebase.html]: This whole block is under indented.

**Comment by ElDeveloper** [comment]
This looks *good* except, there's not a good way to test this code, it has the "same" validity as the other chunk of code taken from a blog post found on the inter-tubes and that the code is not really maintainable. IMHO, should be ready to merge as soon as the comments I made are addressed but then again ... how can we make sure this new snippet is not introducing any problems that the previous block was accounting for? I would probably advocate for finding the bug in the previous block of code and correct it there.

**Comment by antgonza** [comment]
The issue is that there is no perfect way and more when browsers change the way they report their version, which made the previous one obsolete. I think this is fine as long as we test in all valid browsers.

**Comment by ElDeveloper** [comment]
Ok, sounds good. We should check Safari, Chrome, Opera, Internet
Explorer, Firefox and lynx? With the corresponding mobile versions where
possible?

**Comment by antgonza** [comment]
Up to you. I normally check Safari, Chrome, and Firefox.

**Comment by ElDeveloper** [comment]
Well the only browser that shows the error message is lynx, however the interface is unusable in internet explorer 8.

Just for fun:

screen shot 2015-02-10 at 3 51 53 pm

**Comment by wasade** [comment]
@ElDeveloper, that is a fantastic screenshot

**Comment by GabeAl** [comment]
Still no good. I'm being inappropriately flagged (Firefox 35.0.1)
baddetect
 and cannot access the site. I also do not use Chrome and do not run a Mac. Firefox is the only truly open, cross-platform, non-corporate browser that enjoys modern standards support. For Linux (and Windows) users who can't (or won't) use Chrome, it's probably better to botch the detection and include an "unsupported browser" blurb at the top than to blanket ban. 
Cheers

**Comment by antgonza** [comment]
Are you testing on this branch or on the main site?

**Comment by GabeAl** [comment]
Apologies -- actually, this is for the main site as it currently exists, not the proposed changes, thereby merely corroborating the finding in the opening message.

**Comment by squirrelo** [comment]
So apparently flake8 updated and is throwing all sorts of fun errors. I'll fix that in aseperate emergency pull request.

**Comment by ElDeveloper** [comment]
I'm on it, see #846

**Comment by ElDeveloper** [comment]
almost done also ...

**Comment by squirrelo** [comment]
Oops, pull request already in.

**Comment by antgonza** [comment]
Pull from upstream master.

**Comment by squirrelo** [comment]
ready for review again.

**Comment by antgonza** [comment]
:+1: @ElDeveloper ?

**Comment by ElDeveloper** [comment]
It still doesn't work in IE 8 i. e. no message about the website not working is shown. :confused:

**Comment by squirrelo** [comment]
good to review again

**Comment by ElDeveloper** [comment]
It now shows the appropriate message for IE8, however it shows the
message in IE10 and Opera (it wasn't before), from the displayed message
it seems that at least IE10 is supported.

**Comment by squirrelo** [comment]
Bluh, The issue is that the older browsers flat out won't run the version check javascript, so I defaulted it to just throw the error by default. Not sure whats going on with IE 10, but we don't support Opera so it's getting the default error page.

**Comment by ElDeveloper** [comment]
So then the only open issue is IE 10. And we may also need to rephrase
the error message to: "This site only works on these browsers".

**Comment by squirrelo** [comment]
Yup. IE isn't firing off the document.ready so I'll dig into that.

**Comment by squirrelo** [comment]
Went old school and used `body onload=''` to fire off the check, since that's supported by pretty much everything. This also means the site is visible by default now for non-checked browsers, so Opera will show the site again.

**Comment by ElDeveloper** [comment]
If we don't support Opera should that be changed then?

**Comment by squirrelo** [comment]
I think we can have in the documentation that we support X browsers above Y version, and while the site is accessible with other browsers , we make no claims to usability.

**Comment by antgonza** [comment]
Agree, could you add that list to the help wiki?

**Comment by antgonza** [comment]
@ElDeveloper could you check again that it works on the expected browsers? Seems like you have everything in your system to do this ... thanks!

**Comment by squirrelo** [comment]
Added to the FAQ section.

**Comment by antgonza** [comment]
Thanks, looks fine:
https://github.com/biocore/qiita/wiki/Frequently-Asked-Questions

File: qiita_pet/templates/sitebase.html [hunk 3]
```diff
@@ -24,69 +24,41 @@
...
-        if ((objbrowserName == "Chrome"  && objBrMajorVersion < {{allowed_min_browser_versions['chrome']}}) ||
-            (objbrowserName == "Firefox" && objBrMajorVersion < {{allowed_min_browser_versions['firefox']}}) ||
-            (objbrowserName == "Safari"  && objBrMajorVersion < {{allowed_min_browser_versions['safari']}})) {
-          overlay();
-        }
-
-      });
-    </script>
+      }
+
```
Comment (ElDeveloper) [inline qiita_pet/templates/sitebase.html]: Extra whitespace.

**Comment by ElDeveloper** [comment]
Just tested this in all the browsers I have access to and it works just
fine! Lynx actually lets me log in (unlike IE8). Just one more minor
comment about whitespace and this should be merged right away!

**Comment by squirrelo** [comment]
whitespace removed, merge ready.

**Pull request merged**